### ISSUE TITLE
Monitor reports on old certificate being expired, not renewed one

### ISSUE THREAD

**Issue by srgvg** [louislam/uptime-kuma#514]
**Is it a duplicate question?**
No similar open issues.

**Describe the bug**
I have some monitors on https endpoints. Those endpoints use the same certificate which originally recently (yesterday) expired, but that certificate was replaced by a more recent one some weeks ago already.

Somehow that old certificate is being cached, and the monitor reports on the preevious one's expiry date, not taking into account the certificate was replaced already. The cached certificate is not being invalidated.

**To Reproduce**
I didn't try to reproduce it, as that would take several days to reproduce manually, or months to wait for another issue to happen with e.g. ACME certificates.

**Expected behavior**
The monitor picks up the renewed certificate, and keeps reporting it as up, after the old certificate expired.

**Info**
Uptime Kuma Version: 1.7.1
Using Docker?: Not Docker but containerd within Kubernetes (K3S)
Docker Version: K3S 1.20
Node.js Version (Without Docker only): -
OS: K3OS
Browser: Firefox

**Screenshots**
![20210930103159](https://user-images.githubusercontent.com/382239/135417836-b34e57fb-2c1f-421d-b396-2927bfd8c472.png)

In the screenshot you see how this monitor is still reporting down, with an expired certificate, but at the same time, the certificate expiry is shown as "(2021-12-06) 67 days".

On a side note: this host is not really "down". A feature request might be to make a difference in reporting it as down, or just an expired certificate?

**Error Log**
Nothing explicit about this isuue to be found in the log.

**Comment by srgvg** [comment]
I tried clearing data, first events, then heartbeats, but that didn't change anything. Strangely, I never notice anything being gone for that monitor. Also, deleting data seems to trigger some bug for me, where stacktraces are all over the log. This seems solved after restarting the container. I didn't keep those logs yet, so can't tell more about this. Known issue, or should I file an issue next time I see this?

**Comment by chakflying** [comment]
Unfortunately currently the events list does not add a new entry if the host is still "Down" but the error message changed. Can you confirm the latest "Down" message by mouse-over the last heartbeat?

![image](https://user-images.githubusercontent.com/3271800/135432865-7cba487c-4944-45b5-8cd6-264fd1977d8e.png)

**Comment by Oaktribe** [comment]
Same issue here. Tried clearing events and heartbeats of a monitor and restart the container. But it still reports expired certificate. Checking the "Ignore TLS/SSL error for HTTPS websites" options does fix the issue, temporarily.

Can't see anything specific in the logs, more then "Failing: certificate has expired"

**Comment by louislam** [comment]
I heard the news a few week ago.
https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/

I guess @srgvg is using the old ca.pem with a new cert.

**Comment by No0Vad** [comment]
Can confirm same issue here, unsure what caused it because the certificates has been renewed automatically just fine.
But somehow still I was still given certificates with old chain? Either way forcing a renewal fixed it for me too.

For me Uptime-Kuma was actually **quicker** to report something was wrong. Because it still worked on my Windows and Android devices.

**Comment by louislam** [comment]
> It looks to be related to Let's Encrypt certificates. Renewing the certificates fixes it.
https://community.letsencrypt.org/t/r3-intermediate-certificate-has-expired-it-issued-certs-past-its-expiration-date/160797

In the discussion, Someone said:

> The problem is the INTERMEDIATE cert a stated in the subject of this thread.
It isn't updated by certbot

It could be a bug of certbot.

> For me Uptime-Kuma was actually quicker to report something was wrong. Because it still worked on my Windows and Android devices.

As far as I know, Let's encrypt works without intermediate CA on Windows and Android, but not in Node.js.

**Comment by srgvg** [comment]
After checking, it seems my Firefox is still accepting that old root CA. I should have thought of this issue, as the hosts impacted are getting their ACME certirficates from a separate client instance... My apologies.

I'd close this issue now, but perhaps this could be a pointer to a feature request for a better error message explaining why the cert is invalid?

**Comment by louislam** [comment]
The new CA should be expired in 2025. Maybe it is not a urgent task.

**Comment by srgvg** [comment]
As a follow-up, I updated my certificates. Those were indeed specifically still indirectly signed by the older root certificate.

Now, after updating this, I noticed kuma didn't pick them up, and kept returning an expired notice. Until I wen to edit and save (without changing anything), and that triggered it to pick up the new certificate. Suspending then resuming the check also triggered it to pick up the new certificate.

So somehow, this bug report is still valid, it seems?
